src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -339,27 +343,31 @@ private void ForceAuthentication(bool receiveFirst, byte[] buffer)
             }
         }
 
-        internal async Task ForceAuthenticationAsync(bool receiveFirst, byte[] buffer, CancellationToken cancellationToken)
+        internal async Task ForceAuthenticationAsync(bool receiveFirst, byte[] inputData, CancellationToken cancellationToken)
         {
             _framing = Framing.Unknown;
             ProtocolToken message;
-            SslReadAsync adapter = new SslReadAsync(this, cancellationToken);
+            _handshakeBuffer = new ArrayBuffer(InitialReadBufferSize);
```

Comment:
Where is this disposed?

Reply:
I assumed this would be disposed when it goes out of scope. I added explicit Dispose() to ProcessAuthentication() and ReplyOnReAuthenticationAsync(). The first place already had a convenient `finally` block.